src/test/spec/unified_runner/matcher.rs

```diff
@@ -0,0 +1,235 @@
+use crate::bson::{doc, spec::ElementType, Bson};
+
+pub fn results_match(actual: Option<&Bson>, expected: &Bson) -> bool {
+    match expected {
+        Bson::Document(expected_doc) => {
+            if let Some(special_op) = expected_doc.iter().next() {
```

Comment:
Instead of using `iter()` and then `.0`, you can use `expected_doc.keys()`

Reply:
It's still useful to get the entire doc from the iterator since the value gets passed into `special_operator_matches`, but I'll update this to de-structure into `(key, value)`.